File: power-gatling/master/app/routers/system.py
"""System status API routes."""

from fastapi import APIRouter

from ..core import redis as redis_store
from ..core.worker_manager import get_all_workers, get_active_worker_count
from ..models import ApiResponse

router = APIRouter(prefix="/api/system", tags=["system"])


@router.get("/status")
async def system_status():
    redis_connected = False
    try:
        r = await redis_store.get_redis()
        await r.ping()
        redis_connected = True
    except Exception:
        pass

    workers = await get_all_workers()
    active_count = await get_active_worker_count()

    executions = await redis_store.json_list("loadforge:executions:")
    active_executions = [e for e in executions if e.get("status") == "running"]

    return ApiResponse(data={
        "redis_connected": redis_connected,
        "active_workers": active_count,
        "registered_workers": len(workers),
        "active_executions": len(active_executions),
        "workers": workers,
    })


@router.get("/workers")
async def list_workers():
    workers = await get_all_workers()
    return ApiResponse(data=workers)

File: power-gatling/frontend/src/api/system.ts
import api from './client'

export interface SystemStatus {
  redis_connected: boolean
  active_workers: number
  registered_workers: number
  active_executions: number
  workers: Array<{
    worker_id: string
    status: string
    execution_id: string | null
    last_heartbeat: string | null
  }>
}

export const getSystemStatus = () => api.get('/system/status').then(r => r.data) as Promise<SystemStatus>
export const getWorkers = () => api.get('/system/workers').then(r => r.data || [])
